geode-core/src/main/java/org/apache/geode/management/internal/MBeanJMXAdapter.java

```diff
@@ -71,14 +69,17 @@
 
   private DistributedMember distMember;
 
-  private Logger logger = LogService.getLogger();
+  private Logger logger;
```

Comment:
I was able to make all the suggested fields `final` (including the `static final` logger), except for the `mbeanServer`.  I need to be able to replace this server with a mock which simulates a race condition, which is at the heart of the test.  I ended up going with a variation of option #2 above to be able to use the `static final` logger.  We found that option #4 had some side effects when the test is run as a unit test, because there are interactions between Log4J and System.out which prevent running a test in the same JVM twice.